File: scripts/test_solutions/332_reconstruct_itinerary.js
var findItinerary = function(tickets) {
    const graph = new Map();
    for (const [from, to] of tickets) {
        if (!graph.has(from)) graph.set(from, []);
        graph.get(from).push(to);
    }
    for (const [_, dests] of graph) dests.sort().reverse();
    const result = [];
    function dfs(airport) {
        const dests = graph.get(airport);
        while (dests && dests.length) dfs(dests.pop());
        result.push(airport);
    }
    dfs('JFK');
    return result.reverse();
};

File: scripts/test_solutions/332_reconstruct_itinerary.py
from collections import defaultdict

def findItinerary(tickets):
    graph = defaultdict(list)
    for src, dst in tickets:
        graph[src].append(dst)
    for src in graph:
        graph[src].sort(reverse=True)

    result = []

    def dfs(airport):
        while graph[airport]:
            dfs(graph[airport].pop())
        result.append(airport)

    dfs('JFK')
    return result[::-1]
